### DISCUSSION TITLE
Update  app

### DISCUSSION THREAD

**Discussion by leonardo89stg** [NativePHP/laravel#323] [General]
every  i try  update i get this error
PS C:\Users\leona\Desktop\siteprojects\app> php artisan native:publish
PHP Warning:  Module "ftp" is already loaded in Unknown on line 0

Warning: Module "ftp" is already loaded in Unknown on line 0
Building and publishing NativePHP app…

up to date, audited 605 packages in 5s

96 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Warning: Module "ftp" is already loaded in Unknown on line 0
PHP Warning:  Module "ftp" is already loaded in Unknown on line 0
Installing dependencies from lock file
Verifying lock file contents can be installed on current platform.
Package operations: 0 installs, 0 updates, 37 removals
  - Removing theseer/tokenizer (1.2.3)
  - Removing symfony/yaml (v7.1.1)
  - Removing spatie/laravel-ignition (2.7.0)
  - Removing spatie/ignition (1.14.2)
  - Removing spatie/flare-client-php (1.6.0)
  - Removing spatie/backtrace (1.6.1)
  - Removing sebastian/version (4.0.1)
  - Removing sebastian/type (4.0.0)
  - Removing sebastian/recursion-context (5.0.0)
  - Removing sebastian/object-reflector (3.0.0)
  - Removing sebastian/object-enumerator (5.0.0)
  - Removing sebastian/lines-of-code (2.0.2)
  - Removing sebastian/global-state (6.0.2)
  - Removing sebastian/exporter (5.1.2)
  - Removing sebastian/environment (6.1.0)
  - Removing sebastian/diff (5.1.1)
  - Removing sebastian/complexity (3.2.0)
  - Removing sebastian/comparator (5.0.1)
  - Removing sebastian/code-unit-reverse-lookup (3.0.0)
  - Removing sebastian/code-unit (2.0.0)
  - Removing sebastian/cli-parser (2.0.1)
  - Removing phpunit/phpunit (10.5.20)
  - Removing phpunit/php-timer (6.0.0)
  - Removing phpunit/php-text-template (3.0.1)
  - Removing phpunit/php-invoker (4.0.0)
  - Removing phpunit/php-file-iterator (4.1.0)
  - Removing phpunit/php-code-coverage (10.1.14)
  - Removing phar-io/version (3.2.1)
  - Removing phar-io/manifest (2.0.4)
  - Removing nunomaduro/collision (v7.10.0)
  - Removing myclabs/deep-copy (1.11.1)
  - Removing mockery/mockery (1.6.12)
  - Removing laravel/sail (v1.29.2)
  - Removing laravel/pint (v1.16.0)
  - Removing hamcrest/hamcrest-php (v2.0.1)
  - Removing filp/whoops (2.15.4)
  - Removing fakerphp/faker (v1.23.1)
  0/27 [>---------------------------]   0%
 10/27 [==========>-----------------]  37%
 19/27 [===================>--------]  70%
 24/27 [========================>---]  88%
 27/27 [============================] 100%
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

Warning: Module "ftp" is already loaded in Unknown on line 0
PHP Warning:  Module "ftp" is already loaded in Unknown on line 0

   INFO  Discovering packages.

  laravel/sanctum ............................................................................................................................. DONE
  laravel/tinker .............................................................................................................................. DONE
  nativephp/electron .......................................................................................................................... DONE
  nativephp/laravel ........................................................................................................................... DONE
  nesbot/carbon ............................................................................................................................... DONE
  nunomaduro/termwind ......................................................................................................................... DONE

52 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
PHP Fatal error:  Uncaught ErrorException: include(C:\Users\leona\Desktop\siteprojects\app\vendor\composer/../nunomaduro/collision/src/Adapters/Laravel/IgnitionSolutionsRepository.php): Failed to open stream: No such file or directory in C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php:576
Stack trace:
#0 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'include(C:\\User...', 'C:\\Users\\leona\\...', 576)
#1 C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php(576): Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}(2, 'include(C:\\User...', 'C:\\Users\\leona\\...', 576)
#2 C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php(576): include('C:\\Users\\leona\\...')
#3 C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php(427): Composer\Autoload\{closure}('C:\\Users\\leona\\...')
#4 C:\Users\leona\Desktop\siteprojects\app\vendor\nunomaduro\collision\src\Adapters\Laravel\CollisionServiceProvider.php(49): Composer\Autoload\ClassLoader->loadClass('NunoMaduro\\Coll...')
#5 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Container\Container.php(908): NunoMaduro\Collision\Adapters\Laravel\CollisionServiceProvider->NunoMaduro\Collision\Adapters\Laravel\{closure}(Object(Illuminate\Foundation\Application), Array)
#6 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Container\Container.php(795): Illuminate\Container\Container->build(Object(Closure))
#7 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(961): Illuminate\Container\Container->resolve('NunoMaduro\\Coll...', Array, true)
#8 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Container\Container.php(731): Illuminate\Foundation\Application->resolve('NunoMaduro\\Coll...', Array)
#9 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(946): Illuminate\Container\Container->make('NunoMaduro\\Coll...', Array)
#10 C:\Users\leona\Desktop\siteprojects\app\vendor\nunomaduro\collision\src\Adapters\Laravel\ExceptionHandler.php(66): Illuminate\Foundation\Application->make('NunoMaduro\\Coll...')
#11 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(207): NunoMaduro\Collision\Adapters\Laravel\ExceptionHandler->renderForConsole(Object(Symfony\Component\Console\Output\ConsoleOutput), Object(ErrorException))
#12 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(189): Illuminate\Foundation\Bootstrap\HandleExceptions->renderForConsole(Object(ErrorException))   
#13 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(ErrorException))    
#14 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}(Object(ErrorException))
#15 {main}
  thrown in C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php on line 576
PHP Fatal error:  Uncaught ErrorException: include(C:\Users\leona\Desktop\siteprojects\app\vendor\composer/../nunomaduro/collision/src/Adapters/Laravel/IgnitionSolutionsRepository.php): Failed to open stream: No such file or directory in C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php:576
Stack trace:
#0 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'include(C:\\User...', 'C:\\Users\\leona\\...', 576)
#1 C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php(576): Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}(2, 'include(C:\\User...', 'C:\\Users\\leona\\...', 576)
#2 C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php(576): include('C:\\Users\\leona\\...')
#3 C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php(427): Composer\Autoload\{closure}('C:\\Users\\leona\\...')
#4 C:\Users\leona\Desktop\siteprojects\app\vendor\nunomaduro\collision\src\Adapters\Laravel\CollisionServiceProvider.php(49): Composer\Autoload\ClassLoader->loadClass('NunoMaduro\\Coll...')
#5 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Container\Container.php(908): NunoMaduro\Collision\Adapters\Laravel\CollisionServiceProvider->NunoMaduro\Collision\Adapters\Laravel\{closure}(Object(Illuminate\Foundation\Application), Array)
#6 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Container\Container.php(795): Illuminate\Container\Container->build(Object(Closure))
#7 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(961): Illuminate\Container\Container->resolve('NunoMaduro\\Coll...', Array, true)
#8 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Container\Container.php(731): Illuminate\Foundation\Application->resolve('NunoMaduro\\Coll...', Array)
#9 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(946): Illuminate\Container\Container->make('NunoMaduro\\Coll...', Array)
#10 C:\Users\leona\Desktop\siteprojects\app\vendor\nunomaduro\collision\src\Adapters\Laravel\ExceptionHandler.php(66): Illuminate\Foundation\Application->make('NunoMaduro\\Coll...')
#11 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(207): NunoMaduro\Collision\Adapters\Laravel\ExceptionHandler->renderForConsole(Object(Symfony\Component\Console\Output\ConsoleOutput), Object(Symfony\Component\ErrorHandler\Error\FatalError))
#12 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(189): Illuminate\Foundation\Bootstrap\HandleExceptions->renderForConsole(Object(Symfony\Component\ErrorHandler\Error\FatalError))
#13 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(231): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(Symfony\Component\ErrorHandler\Error\FatalError))
#14 C:\Users\leona\Desktop\siteprojects\app\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleShutdown()
#15 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#16 {main}
  thrown in C:\Users\leona\Desktop\siteprojects\app\vendor\composer\ClassLoader.php on line 576
  Config :
  
  
  <?php

return [
    /**
     * The version of your app.
     * It is used to determine if the app needs to be updated.
     * Increment this value every time you release a new version of your app.
     */
    'version' => env('NATIVEPHP_APP_VERSION', '1.0.1'),

    /**
     * The ID of your application. This should be a unique identifier
     * usually in the form of a reverse domain name.
     * For example: com.nativephp.app
     */
    'app_id' => env('NATIVEPHP_APP_ID', 'com.nativephp.app'),

    /**
     * If your application allows deep linking, you can specify the scheme
     * to use here. This is the scheme that will be used to open your
     * application from within other applications.
     * For example: "nativephp"
     *
     * This would allow you to open your application using a URL like:
     * nativephp://some/path
     */
    'deeplink_scheme' => env('NATIVEPHP_DEEPLINK_SCHEME'),

    /**
     * The author of your application.
     */
    'author' => env('NATIVEPHP_APP_AUTHOR'),

    /**
     * The default service provider for your application. This provider
     * takes care of bootstrapping your application and configuring
     * any global hotkeys, menus, windows, etc.
     */
    'provider' => \App\Providers\NativeAppServiceProvider::class,

    /**
     * A list of environment keys that should be removed from the
     * .env file when the application is bundled for production.
     * You may use wildcards to match multiple keys.
     */
    'cleanup_env_keys' => [
        'AWS_*',
        'GITHUB_*',
        'DO_SPACES_*',
        '*_SECRET',
        'NATIVEPHP_UPDATER_PATH',
        'NATIVEPHP_APPLE_ID',
        'NATIVEPHP_APPLE_ID_PASS',
        'NATIVEPHP_APPLE_TEAM_ID',
    ],

    /**
     * A list of files and folders that should be removed from the
     * final app before it is bundled for production.
     * You may use glob / wildcard patterns here.
     */
    'cleanup_exclude_files' => [
        'content',
        'storage/app/framework/{sessions,testing,cache}',
        'storage/logs/laravel.log',
    ],

    /**
     * The NativePHP updater configuration.
     */
    'updater' => [
        /**
         * Whether or not the updater is enabled. Please note that the
         * updater will only work when your application is bundled
         * for production.
         */
        'enabled' => env('NATIVEPHP_UPDATER_ENABLED', true),

        /**
         * The updater provider to use.
         * Supported: "github", "s3", "spaces"
         */
        'default' => env('NATIVEPHP_UPDATER_PROVIDER', 'github'),

        'providers' => [
            'github' => [
                'driver' => 'github',
                'repo' => env('GITHUB_REPO','app'),
                'owner' => env('GITHUB_OWNER','leonardo89stg'),
                'token' => env('GITHUB_TOKEN','ssdsd'),
                'vPrefixedTagName' => env('GITHUB_V_PREFIXED_TAG_NAME', true),
                'private' => env('GITHUB_PRIVATE', false),
                'channel' => env('GITHUB_CHANNEL', 'latest'),
                'releaseType' => env('GITHUB_RELEASE_TYPE', 'draft'),
            ],

            's3' => [
                'driver' => 's3',
                'key' => env('AWS_ACCESS_KEY_ID'),
                'secret' => env('AWS_SECRET_ACCESS_KEY'),
                'region' => env('AWS_DEFAULT_REGION'),
                'bucket' => env('AWS_BUCKET'),
                'endpoint' => env('AWS_ENDPOINT'),
                'path' => env('NATIVEPHP_UPDATER_PATH', null),
            ],

            'spaces' => [
                'driver' => 'spaces',
                'key' => env('DO_SPACES_KEY_ID'),
                'secret' => env('DO_SPACES_SECRET_ACCESS_KEY'),
                'name' => env('DO_SPACES_NAME'),
                'region' => env('DO_SPACES_REGION'),
                'path' => env('NATIVEPHP_UPDATER_PATH', null),
            ],
        ],
    ],
];

**Comment by simonhamp** [comment]
This looks more like an issue than a Discussion. Please [raise an issue](https://github.com/nativephp/laravel/issues/new/choose)

**Reply by leonardo89stg** [reply]
ok ! :)
